envi/const.py

```diff
@@ -18,6 +18,7 @@
 MM_WRITE = 0x2
 MM_EXEC = 0x1
 MM_SHARED = 0x08
+MM_UNINIT = 0x10
```

Comment:
what is this exactly?
for ELF, is this like BSS or something similar?

this all looks pretty good.

i'm confused tho.  this is badbytes.  should we handle something along the same lines?  or is this the best way for Viv/Codeflow?

at one point, @invisig0th and i were talking about doing away with the CodeFlow subsystem and going straight Emulation for all of the recursive descent.  
the reason this didn't happen, in my opinion, was 1) time... and 2) the fact that it requires that all architectures support a specific set of instruction emulation to perform the recursive descent.

this also opens the conversation of self-modifying code handling as an analysis module... probably something pretty early in the analysis process.

Reply:
It can be something like that for linux. I just wanted to discuss that part.  Right now I'm just focusing in the PE file at hand that's causing an issue.

>  this is badbytes 

Well, sort of? The section being jumped to while we're in codeflow is uninitialized data. The actual instructions beforehand we get are correct. We just haven't modified the workspace code to do stuff for unpacking. So until we do have that, if we want to avoid issues like this, my first thought was to a). lift up knowledge of the section being uninitialized from the PE file, and b). catch that flag so that codeflow doesn't fall down into there and spin off into oblivion like it's doing now. I'm open to a different approach, but this occurs well before full workspace emulation while we're in codeflow, so we kind of have to handle it there, since codeflow is the thing running off into the void right now.

So yea, it does open up the longer epic of dealing with self-modifying code. I feel like that's a much, much larger task than making viv just not crash here on jumping to section with a known flag we can catch and handle. That should be something we talk about and design on its own rather than in the scope of fixing an issue.

I've had the thought of ripping out codeflow for a while now too. It's just taken a back seat to everything else I want to get into viv. It'd be a massive change to a lot of things that isn't as fully formed in my head of "I have to do this" as finishing some of the open issues and dwarf and what not are.

Side note: this also brings up the whole mess of dealing with recursive hydra functions again, since that's why the CI tests are now failing. So I'm going back to that well now too. So there might be some scope creep on this PR.

Reply:
i'm on board with all of this.
i just didn't want to miss the opportunity to spin the conversation up :)

if the CI weren't failing, i'd approve this.

Reply:
That's fair.

Though CI is doing its job, so don't be too hard on it :D Those weird recursive functions strike again so I'm gonna take another try at those to see if we can get something working there.